### ISSUE TITLE
[BUG] Negative margins are not Tailwind classes

### ISSUE THREAD

**Issue by Serator** [francoismassart/eslint-plugin-tailwindcss#78]
**Describe the bug**

When you insert the `-mt-8` class, you get an invalid warning:
```
Classname '-mt-8' is not a Tailwind CSS class! | eslinttailwindcss/no-custom-classname
```
This applies to all negative margins (`-mx-*`, `-my-*`, `lg:-mb-*`...).

**To Reproduce**
1. Type `-mt-8` in your code.
2. Run lint
3. You will get the message.

**Expected behavior**
No warning would happen.

**Environment (please complete the following information):**
 - OS: Windows 10
 - Softwares + version used:
     -  VSCode: 1.63.2
     - eslint-plugin-tailwindcss: 3.0.0
     - tailwindcss: 3.0.7

**Comment by francoismassart** [comment]
Strange, I added the following test as valid and it passes:

```
    {
      code: `
      <div className={'-mt-8'}>Negative margin</div>`,
    },
```

I'll try it in a real project, to see if I still get it.

Did you try:

1. Removing all your `node_modules/` folder
2. `npm i`
3. Restart VSCode

Thank you for your feedback

**Comment by francoismassart** [comment]
I did use negative margin in a real project with the beta0 of version 3 and it did not complain about it...

I'll be waiting for your feedback @Serator 😉

**Comment by Serator** [comment]
@francoismassart Thank you for a great package and a super quick response!

I tried to isolate the problem and it is still reproducing.

`tailwind.config.js`
```
module.exports = {
  content: ['./src/**/*.tsx'],

  theme: {
    spacing: {
      4: 'calc(4 * .25rem)',
    },
  },
}

```

`.eslintrc.js`
```
module.exports = {
  parser: '@typescript-eslint/parser',
  plugins: ['tailwindcss'],
  rules: {'tailwindcss/no-custom-classname': ['warn']},
}
```

* @typescript-eslint/parser: 5.6.0
* eslint: 8.4.1
* eslint-plugin-tailwindcss: 3.0.1
* tailwindcss: 3.0.8

Test code:
```
import {memo} from 'react'

export const Test = memo(function Test() {
  return <div className="-mt-4" />
})
```

![image](https://user-images.githubusercontent.com/625005/147611087-54ed0e13-6d72-4f4c-a9b4-2968fcf0d1e4.png)

**Comment by Serator** [comment]
@francoismassart I did some more research on the problem. The error occurs when `spacing` values are overwritten in `tailwind.config.js` (example in previous post). There is no error if you [test](https://github.com/francoismassart/eslint-plugin-tailwindcss/blob/2fed34c1b0adde8f9055ff303736ab41dce28175/tests/lib/rules/no-custom-classname.js#L145) with default Tailwind configuration.

**Comment by francoismassart** [comment]
@Serator, thank for your investigation, I'm now able to reproduce the issue. WIP

**Comment by kevinmu17** [comment]
Seems to be back in v3.1.3
